Makefile

```diff
@@ -179,6 +179,15 @@ endif
 CFLAGS += -MMD -MP -MF"$(@:%.o=%.d)"
 
 
+#######################################
+# edge case
+#######################################
+# exclude CAN
+ifeq ($(filter $(SERIES_LINE_GENERIC), stm32f401xe stm32f401xc), $(SERIES_LINE_GENERIC))
+    C_SOURCES := $(filter-out driver/Src/CAN.c, $(C_SOURCES))
+    C_DEFS += -DCAN_UNDEFINED
+endif
+
```

Comment:
oops we need to make this bsp/Src/CAN.c now

Reply:
also in future we should just have some feature flag macro for the port that enables CAN/SPI/whatever